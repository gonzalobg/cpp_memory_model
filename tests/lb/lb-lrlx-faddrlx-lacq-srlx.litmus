C lb-lrlx-faddrlx-lacq-srlx
{ [x] = 0; [y] = 0;  }

P0 (int* x, int* y) {
  int b = atomic_load_explicit(y, memory_order_relaxed);
  atomic_fetch_add_explicit(x, 1, memory_order_relaxed);
}

P1 (int* x, int* y) {
  int a = atomic_load_explicit(x, memory_order_acquire);
  if (a == 1) {
    atomic_store_explicit(y, 1, memory_order_relaxed);
  }
}

exists(0:b=1)