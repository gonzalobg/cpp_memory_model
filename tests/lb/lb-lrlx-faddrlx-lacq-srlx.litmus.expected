Test lb-lrlx-faddrlx-lacq-srlx Allowed
States 2
0:b=0;
0:b=1;
Ok
Witnesses
Positive: 1 Negative: 2
Condition exists (0:b=1)
Observation lb-lrlx-faddrlx-lacq-srlx Sometimes 1 2
Hash=ea3d9c46a25a9603860ba98d1b1086b7

